tests/integration/galley/webhook_test.go

```diff
@@ -0,0 +1,103 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package galley
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/environment"
+	"istio.io/istio/pkg/test/framework/components/environment/kube"
+)
+
+var (
+	vwcName    = "istio-galley"
+	deployName = "istio-galley"
+	istioNs    = "istio-system"
+	sleepDelay = 3 * time.Second // How long to wait to give the reconcile loop an opportunity to act
+)
+
+var removeGalleyYaml = fmt.Sprintf(`
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: %s
+  namespace: %s
+`, deployName, istioNs)
+
+// Verify that scaling up/down doesn't modify webhook configuration
+func TestWebhookScaling(t *testing.T) {
+	framework.NewTest(t).
+		// Limit to Kube environment as we're testing integration of webhook with K8s.
+		RequiresEnvironment(environment.Kube).
+		Run(func(ctx framework.TestContext) {
+			env := ctx.Environment().(*kube.Environment)
+			startGen := getVwcGeneration(vwcName, t, env)
+
+			// Scale up
+			scaleDeployment(istioNs, deployName, 2, t, env)
+			// Wait a bit to give the ValidatingWebhookConfiguration reconcile loop an opportunity to act
+			time.Sleep(sleepDelay)
+			gen := getVwcGeneration(vwcName, t, env)
+			if gen != startGen {
+				t.Fatalf("ValidatingWebhookConfiguration was updated unexpectedly on scale up")
+			}
+
+			// Scale down
+			scaleDeployment(istioNs, deployName, 1, t, env)
+			// Wait a bit to give the ValidatingWebhookConfiguration reconcile loop an opportunity to act
+			time.Sleep(sleepDelay)
```

Comment:
This should be in a retry loop to avoid test flakes. There is a retry package for doing this.

Reply:
The scaleDeployment has retry involved since it's changing the number of replicas and then waiting (with retry) for the state to match. 

I'm not sure how a retry can replace the sleep, because what I'm doing is sort of inverted from the usual case. I'm not waiting for a condition to trigger, I'm waiting a bit to make sure it does *not* trigger to avoid a false pass on the test if there's a race condition. I'm open to ideas on how to handle this better.

Reply:
Ugh. Yup you're right.